### ISSUE TITLE
[Bug]: Sort/Freeze controls still exist in the Table header after sorting and freezing have both been disabled.

### ISSUE THREAD

**Issue by AS-Laguna** [appsmithorg/appsmith#23200]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The sort/freeze controls still exist in the DOM if sorting and freezing are disabled for a column. This causes unnecessary squeeze on the column headers.

I expect the DOM element to disappear and give way to the header text if both behaviors are disabled.

Here is an example of this squeeze. If the user makes one of the columns any smaller, the header text is truncated.

![image](https://github.com/appsmithorg/appsmith/assets/101155659/9daef3e1-15a9-411c-8313-2779d00ef799)

### Steps To Reproduce

1. D&D a table onto the canvas.
2. Disable sorting and freezing for the table.
3. Observe that the dropdown arrow still exists in the header.
4. Reduce the width of one of the columns.
5. Observe that this control takes up space and causes premature truncation of the header text.

### Environment

Production

### Version

Cloud v1.9.18-SNAPSHOT

**Comment by pchambless** [comment]
This will help the mobile App I'm developing immensely with Real Estate.  Thanks!

**Comment by Sheikh-JamirAlam** [comment]
@somangshu @dilippitchika Can I work on this issue?

**Comment by dilippitchika** [comment]
@Sheikh-JamirAlam yes sure, assigning this to you. Feel free to raise a PR in the next few days.

**Comment by Sheikh-JamirAlam** [comment]
@AS-Laguna @dilippitchika Should I raise a PR?

[Screencast from 02-06-23 11:59:25 AM IST.webm](https://github.com/appsmithorg/appsmith/assets/48979376/bc57766e-4e64-4ae5-93e0-6a184f06bc81)

**Comment by Sheikh-JamirAlam** [comment]
Please review the PR @AS-Laguna @dilippitchika
